### ISSUE TITLE
[Question] new version?

### ISSUE THREAD

**Issue by gilbh** [pit-ray/win-vind#124]
Hello,

I have been working with your amazing tool for a long time, following announces etc., but I want to make sure I did not miss any recent release. If I am not mistaken, a lot of new features and bug corrections were implemented since the last release of 4.3.3, which was a long time ago (maybe a year?).

Just thought of asking about this. N pressure intended, as your contributions are greatly appreciated.

Best

**Comment by pit-ray** [comment]
As you say, win-vind has not been released since February of this year and I could not respond to many issues or feedback recently.

Sorry.

I'm so busy this year with research, so I commit to an active pull request little by little when I'm free. However, win-vind have had enthusiast recently as you, so the win-vind development procedure must be renewed. 


## New plan to develop actively
In the win-vind development procedure the heaviest processes are testing/debugging and document updating, so I released batch-style, releasing after finishing some the feature addition or the bug fix, to minimize these procedures. As the result, the release of win-vind had stopped as opposed to what I had said "next version".

However, this procedure is not good due to clogging if some implementations are heavy(like now). Therefore, I plan to adapt task-style releasing instead of batch-style. In this style, all users are available with new features or fixed functions right now, but having heavy testing and frequent document updating as the downside. In other words, the releasing speed is faster, but the developing becomes slow. 

If some contributor helps to test new features or to write documents for this project, the cons are solved. To join this project easily, I'll use Projects, which is a new feature of GitHub, instead of milestones or the simple TODO list. Because the conventional ways hide the active issues and are difficult to understand. Furthermore, I take in one or two collaborators to want to join this project. The debugging team tests the finished task and reports whether the change causes new issues with the easy-to-join black-box unit test.

Please give your thinking for this strategy.


## Upcoming
As seen in the many issues, the current win-vind causes many problems due to the mapping system or hard-understanding .vindrc syntax rather than the feature unit issue.

So, redesigning the core of win-vind is most important. I have finished 80% of the implementation currently and should install it into each function to fit the new core system from now on. However, this change causes a very large impact on the all codes, so I estimate it to finish in about two or three weeks.

If the collaborator joins and tests this master branch, the most recent release is completed right now. If not then, the release after this implementation. I'm just one.

Thanks.

**Comment by pit-ray** [comment]
Currently, I'm listing tasks to be done in https://github.com/users/pit-ray/projects/2.

**Comment by gilbh** [comment]
Thank you.

I think the projects is a good arrangement for tracking progress and anticipating new releases. 

I am not sure I understand your point about batch vs. speed of release. I can't code for win-vind but if you want me to help with release/documentation I could try. I am not experienced in github though.

Best

**Comment by pit-ray** [comment]
@gilbh 
Some modifications in the master branch assume for this latest my pull-request, so I will notify your help wanted after some updates.

Thanks for your many contributions.

**Comment by pit-ray** [comment]
@gilbh 
I merged changes #109 to current master branch. As soon as the obvious bugs are fixed, I will call for beta testing to be started in Discussion.

**Comment by gilbh** [comment]
Thanks. I will be here to check!!

**Comment by pit-ray** [comment]
@gilbh 
Please check #129.

**Comment by gilbh** [comment]
Fantastic and congrats. I will get to this.

**Comment by gilbh** [comment]
Okay, 

I am ready to start testing the new version. I installed version 5.0.0 and managed to get it to run. But my .vindrc is not working anymore. This is because the mapping conventions has changed, but I can't figure out what exactly change and how to upgrade the .vindrc. 

Can you pleas advise on the following? With many thanks!
```
set suppress_for_vim = true
set autofocus_textarea = true

set cursor_accel = 2
set cursor_maxv = 1
set cursor_tweight = 50

" set easyclick_bgcolor=#cb9925
set easyclick_bgcolor=#99cc00
set easyclick_fontcolor=#000000
set easyclick_fontweight=900  " maximum value is 1000
set easyclick_fontsize=18
set easyclick_fontname=Arial

" Virtual command line options
set shell = cmd
set cmd_fontsize = 14
set cmd_roughpos = LowerLeft
set cmd_maxchar = 100
set cmd_fadeout = 3600

" Enable block style caret
set blockstylecaret
set blockstylecaret_mode = flex
set blockstylecaret_width = 1

" SPECIFIC MODES
" ==============

" from INSERT
inoremap <f15> switch_window
inoremap <insert> to_edi_normal
inoremap <esc> to_resident
inoremap <f20> easy_click_left
inoremap <ctrl-8> <f20>
inoremap <ctrl-6> :set cmd_fadeout=2<cr><insert>

" from EDIT NORMAL
ennoremap <alt-tab> i<f15>
ennoremap <tab> move_caret_right
ennoremap <shift-tab> move_caret_left
ennoremap <insert> to_insert
ennoremap <esc> to_resident
ennoremap w move_fwd_word_simple
ennoremap b move_bck_word_simple
ennoremap <f19> to_resident
ennoremap t i<f20>
ennoremap <ctrl-6> :set cmd_fadeout=2<cr><insert>
enmap / <f19><ctrl-f>

" from EDIT VISUAL
evnoremap i to_insert
evnoremap <insert> to_edi_normal
evnoremap <esc> to_resident
evnoremap t i<f20>
evnoremap <ctrl-6> :set cmd_fadeout=2<cr><insert>
evnoremap <alt-i> to_insert<alt-i>

" from GUI NORMAL
gnnoremap <insert> to_edi_normal
gnnoremap <esc> to_resident
gnnoremap t i<f20>
gnnoremap <ctrl-6> :set cmd_fadeout=2<cr><insert>

" from RESIDENT
rnoremap <insert> to_edi_normal
rnoremap <ctrl-6> :set cmd_fadeout=2<cr><insert>
```

**Comment by pit-ray** [comment]
I'll check it.

**Comment by pit-ray** [comment]
@gilbh 
Since I'm busy at the end of the year, sorry for the late reply.

Please read the release notes more.
And then, if you are hard to understand, I add the details about that points.

So goes the main part, I advise the missing point to your .vindrc.

First, some parameters are eliminated and introduced their replacements in v5. About these points, please read the **Eliminated options and replacements** in [release notes](https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta1).
In your .vindrc, the `cursor_tweight` is replaced with `cursor_resolution` and `cursor_maxv` are eliminated. However, the `cursor_tweight` should also be supported in the production release, but the beta does not support it yet.

Second, the function name must enclose in `<>` brackets, so the `ennoremap <tab> move_caret_right` should become `ennoremap <tab> <move_caret_right>`. In this point, please read the **Arguments of map/noremap** in [release notes](https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta1).

Third, the difference between `noremap` and `map` is fundamentally changed, so please read the head section **The difference between map and normap** in [release notes](https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta1). `enmap / <f19><ctrl-f>` should be `enmap / <f19>{<ctrl-f>}`. And the `noremap` does not refer to the user-defined map, so the `enoremap <alt-tab> i<f15>` does not refer `inoremap <f15> <switch_window>`.

I hope I can be of help.

**Comment by gilbh** [comment]
Thank you for the detailed explanation. I will get to this soon and let you know how it went.

**Comment by pit-ray** [comment]
@gilbh 
Check out the beta2 just released.
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta2

**Comment by gilbh** [comment]
Great. A quick change of adding `<` `>`  to all the commands (e.g., `to_insert`) worked fine. I would need to look deeper to know if there are any problems, but right now the major operations (like switching modes with my remaps) seem to work fine!

**Comment by gilbh** [comment]
Okay, something broke with my configurations and nothing works now with v5.0.0.5. Can you please take a look at this and let me know what I am doing wrong? With many thanks in advance.
[vindrc.txt](https://github.com/pit-ray/win-vind/files/10530045/vindrc.txt)

**Comment by pit-ray** [comment]
The file name must be `.vindrc`, not `vindrc.txt`.
Also, your .vindrc don't have problems in my environment, what exactly is not working?

**Comment by gilbh** [comment]
Thanks. The file name on my computer is `.vindrc`. I renamed it so github will allow for it to upload.

None of the keys (such as `<insert>`) are working now.

**Comment by pit-ray** [comment]
@gilbh 
> Thanks. The file name on my computer is .vindrc. I renamed it so github will allow for it to upload.  

I understand.

> None of the keys (such as <insert>) are working now.

The `<insert>` not working is the mapping that calls `<to_edi_normal>`, right? Can you elaborate a bit more on the fact that it doesn't work?

To debug, unzip the portable version and please try to investigate in a minimal environment.

Thanks.

**Comment by gilbh** [comment]
yes, that's is the problem. I will get back to you on this after testing.

**Comment by pit-ray** [comment]
This Issue is closed as 5.0.0 has been released. If you have any problems, please start a new issue.
